### PR TITLE
Lesson/basic enumerable methods

### PR DESCRIPTION
**Pull request by PolarisTLX**
Related to Issue #112 Ruby Basics Lessons.

We have created the draft for the lesson on Basic Enumerable Methods. Let us know if you have any feedback or requested changes.

Co-authored-by: Kyle Lemon <james.kyle.lemon@gmail.com> and Paul Rail <paul.a.rail@gmail.com>

### CODE REVIEW

**Comment by PolarisTLX** [comment]
One thing we are uncertain about, is why our commits appear to be doubled.  We believe this occurred while we were trying to make sure our fork was rebased / up-to-date to the main repo before submitting the pull request.

**Comment by codyloyd** [comment]
guys this sucks what are you doing...








...totally kidding.   My ruby expertise is pretty minimal so I'll let one of the other guys review it properly. I just wanted to drop in and say THANK YOU SO MUCH for the effort that went into this.  Really great job here.  Stuff like this is what keeps us going!

File: new_course/1_programming_fundementals/basic_enumerable_methods.md [hunk 1]
```diff
@@ -1,51 +1,343 @@
-# Basic Enumerable Methods
-* what enumerable methods are
-* some can be used on hashes, others turn a hash into an array
+## Introduction
+
+The Enumerable module is a set of methods that is included in certain Ruby classes (like Array and Hash). You can even include it in your own classes, which will be covered later on. These methods are handy for looping through iterable objects in a collection (most commonly arrays, hashes, and ranges). They provide
```
Comment (KevinMulhern) [inline new_course/1_programming_fundementals/basic_enumerable_methods.md]: Modules and classes are something the student won't have been exposed to by this part of the course yet. Those will be the main topics in the section that comes after this section of lessons.

What would you think about changing the first line to something like this instead
```
Enumerables are a set of methods that are available on certain collections in ruby (like Array and Hash)
```

File: new_course/1_programming_fundementals/basic_enumerable_methods.md [hunk 2]
```diff
@@ -1,51 +1,343 @@
...
+There is a lot here, but these are built-in to make your life easier. We will run you through the ones that you will most commonly use. This certainly is not an exhaustive list. At the end of this lesson there is a link for you to find out more about other methods beyond what we go through here.
+
+It will be beneficial for you to code along to test the examples as you work through this lesson (in IRB).
+<br/>
 
 ## Learning outcomes
 *Look through these now and then use them to test yourself after doing the assignment*
 
-what the student is expected to know or be able to do by the end of this lesson
+* What kings of objects can `enumerable` methods be called on?
```
Comment (KevinMulhern) [inline new_course/1_programming_fundementals/basic_enumerable_methods.md]: Looks like a typo here "kings"

File: new_course/1_programming_fundementals/basic_enumerable_methods.md [hunk 3]
```diff
@@ -1,51 +1,343 @@
...
-* what enumerable methods are
-* some can be used on hashes, others turn a hash into an array
+## Introduction
+
+The Enumerable module is a set of methods that is included in certain Ruby classes (like Array and Hash). You can even include it in your own classes, which will be covered later on. These methods are handy for looping through iterable objects in a collection (most commonly arrays, hashes, and ranges). They provide
+easy ways to perform common actions, and are among the most important tools for Rubyists. Some enumerable methods can be used on hashes just as on arrays, while others are implemented differently depending on the type of collection used.
+
+There is a lot here, but these are built-in to make your life easier. We will run you through the ones that you will most commonly use. This certainly is not an exhaustive list. At the end of this lesson there is a link for you to find out more about other methods beyond what we go through here.
+
+It will be beneficial for you to code along to test the examples as you work through this lesson (in IRB).
```
Comment (KevinMulhern) [inline new_course/1_programming_fundementals/basic_enumerable_methods.md]: What do you think about adding an additional link here to repl.it as an alternative to IRB? just to give more options to ensure the student is more likely to follow along.

**Comment by KevinMulhern** [comment]
Instead of leaving a bunch of comments about this in the review I'll just leave one here instead. We're using the word "object" in a few different places within the lesson. 

for example in the paragraph about each:
```
Calling the `each` method on an iterable object will loop through each item in that object and perform a task, which you define in a code block that you state after calling `each`.
```

The student will probably have been exposed to "Everything in Ruby is an object" before this lesson. But I'm not sure if they will understand what it actually means until they go through the Basics of Object Oriented Programming section which will come after the section this lesson be part of.

Do you think we should swap out the use of "object" within this lesson with something more explicit? for example"array", "hash"  or "collection" depending on the context.

It's not something that I'm massively bothered about, its just something I noticed and though it would be worth discussing. If you guys think object is fine, thats good enough for me too 😄

**Comment by KevinMulhern** [comment]
I want to echo what @codyloyd has said by thanking you so much for all the work and effort you guys have put into this lesson. The quality of the content is outstanding and it the flow from section to section is sooo good.

Other than the few minor comments I've left, this lesson is perfect and will be ready to merge once they are addressed. Let me know if you have any questions about them please 😄 

I can't wait to get this lesson live, its such a massive improvement over what we have now. Really great work 👍

**Comment by KevinMulhern** [comment]
Thanks for implementing the feedback guys. And thanks again for the great work you guys have done with lesson.

👍 LGTM

**Pull request merged**